src/python/pants/invalidation/build_invalidator.py

```diff
@@ -93,8 +94,30 @@ def key_for_target(self, target, transitive=False, fingerprint_strategy=None):
 class BuildInvalidator(object):
   """Invalidates build targets based on the SHA1 hash of source files and other inputs."""
 
-  def __init__(self, root):
-    self._root = os.path.join(root, GLOBAL_CACHE_KEY_GEN_VERSION)
+  class Factory(Subsystem):
+    options_scope = 'build-invalidator'
+
```

Comment:
Using a `--cache-ignore` option instead, so this is still a subsystem that exports no options, just reads them, and the oddness remains, but seems fine to me. Let me know if this is a big deal.